src/expr.cpp

```diff
@@ -5312,53 +5403,55 @@ ConstExpr::ConstExpr(const Type *t, bool *b, SourcePos p) : Expr(p, ConstExprID)
         boolVal[j] = b[j];
 }
 
-ConstExpr::ConstExpr(ConstExpr *old, double *v) : Expr(old->pos, ConstExprID) {
+// The llvm::APFloat value is guaranteed to be a double.
+ConstExpr::ConstExpr(ConstExpr *old, std::vector<llvm::APFloat> v) : Expr(old->pos, ConstExprID) {
     type = old->type;
 
     AtomicType::BasicType basicType = getBasicType();
 
     switch (basicType) {
     case AtomicType::TYPE_BOOL:
         for (int i = 0; i < Count(); ++i)
-            boolVal[i] = (v[i] != 0.);
```

Comment:
This is an interesting case. Old behavior and new behavior are different - comparing for equality will succeed only for `+0.0`, but not for `-0.0`. While new version will handle both values. I'm not sure what is correct behavior. I guess old behavior is more correct, as it will match runtime behavior. Would be good how C/C++ compiler behave.

Reply:
That part can actually be further simplified. Will upload a change. But to confirm behavior - https://godbolt.org/z/MPMo1shx6

Reply:
I've modified it since that line of code would not have been hit anyway.

Reply:
Mu comment here was incorrect, According to IEE754, comparisons shall ignore the sign of zero (so +0 == −0).